bpf/tc/attach.go

```diff
@@ -140,6 +153,9 @@ func (ap AttachPoint) AttachProgram() error {
 		return fmt.Errorf("failed to clean up one or more old calico programs: %v", progErrs)
 	}
 
+	if err = bpf.RememberAttachedProgs(ap.IfaceName(), ap.FileName(), objHash); err != nil {
+		log.Error(err)
```

Comment:
Again, this isn't the right way to use logrus.  The first argument to Error/Info/Debug should always be a string message.  Then you have a choice of using logrus' key/value based logging or a formatted log:
```
logrus.WithError(err).Error("Failed to record hash of BPF program on disk.  Ignoring.")
logrus.Errorf("Failed to record hash of BPF program on disk: %v.  Ignoring.", err)
```
We generally prefer the KV style but sometimes you want more control over the formatting.